src/main/java/org/spoken_tutorial/health/elasticsearch/models/QueueManagement.java

```diff
@@ -411,7 +414,10 @@ else if (getLanguageId() != documentSearch.getLanguageId()) {
                     if (getRequestType().equals(Config.UPDATE_DOCUMENT)) {
                         String path = getDocumentPath();
                         if (path.startsWith("https://")) {
-                            path = jsonService.saveNarrationToFile(path, getDocumentId());
+                            if (taskProcessingService.isURLWorking(path)) {
```

Comment:
This is a wrong place for this check. Check should be done before submitting the task here https://github.com/Spoken-tutorial/HealthNutritionDocuments/blob/5d51f611f03a7f159f0fe49b5f2fc6c5a91b75b3/src/main/java/org/spoken_tutorial/health/elasticsearch/threadpool/TaskProcessingService.java#L90